js/build.js

```diff
@@ -194,22 +210,45 @@
         refreshTimer = setTimeout(refresh, refreshTimeout);
       }
 
-      function inheritColor(inheritanceColorKey, colorsArray, colorIndex) {
-        var inheritanceColor = Fliplet.Themes.Current.get(inheritanceColorKey);
+      function getThemeColor(colorKey) {
+        return (themeValues && themeValues.hasOwnProperty(colorKey)) ? themeValues[colorKey] : Fliplet.Themes.Current.get(colorKey);
+      }
 
-        if (inheritanceColor) {
-          colorsArray[colorIndex] = inheritanceColor;
+      function getDeviceType() {
+        if (Modernizr.mobile) {
+          return '';
+        } else if (Modernizr.tablet) {
+          return 'Tablet';
         }
+
+        return 'Desktop';
       }
 
       Fliplet.Studio.onEvent(function(event) {
         var eventDetail = event.detail;
 
+        if (eventDetail && eventDetail.type === 'savingNewStyles') {
+          if (eventDetail.widgetId && eventDetail.widgetId !== chartId) {
+            return;
+          }
+
+          setThemeValues(eventDetail.data);
+        }
+
         if (eventDetail && eventDetail.type === 'colorChange') {
           if (eventDetail.widgetId && eventDetail.widgetId !== chartId) {
             return;
           }
 
+          var widgetColors = getWidgetColors(themeInstance.data.widgetInstances);
+
+          if (
```

Comment:
This is the format we'd use.

```js
if (a
  && b) {
  return;
}
```

- `&&` operator goes in the beginning of the next line, not the end of a line
- No lone `(` or `)`

Reply:
Thanks! I'll fix this